internal/db/tx.go

```diff
@@ -0,0 +1,143 @@
+package db
+
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrData         = errors.New("data-error")
+	ErrInvalidModel = fmt.Errorf("%w: invalid model", ErrData)
+)
+
+type Tx interface {
+	Schema() *Schema
+	loadFunction(Record) (Function, error)
+
+	getRelatedOne(ID, ID, ID) (Record, error)
+	getRelatedMany(ID, ID, ID) ([]Record, error)
+	getRelatedManyReverse(ID, ID, ID) ([]Record, error)
+	getRelatedOneReverse(ID, ID, ID) (Record, error)
+
+	FindOne(ID, Matcher) (Record, error)
+	FindMany(ID, Matcher) ([]Record, error)
+	Ref(ID) ModelRef
+	Query(ModelRef) Q
+}
+
+type RWTx interface {
+	Schema() *Schema
+
+	// reads
+	getRelatedOne(ID, ID, ID) (Record, error)
```

Comment:
not too sure about these -- wonder if they should get ripped out and let concrete/reverse rel talk to holdTx directly